tests/lib/rules/highest-available-document-mode/tests.ts

```diff
@@ -91,25 +91,46 @@ const testsForRequireMetaTagConfig: Array<IRuleTest> = [
     }
 ];
 
-const testsForTargetBrowsersConfig: Array<IRuleTest> = [
+const testsForHeaderWithBlankTargetBrowsersConfig: Array<IRuleTest> = [
     {
-        name: `HTML page is served with 'X-UA-Compatible' header but the targeted browsers don't support document modes`,
-        reports: [{ message: `'x-ua-compatible' header is not needed` }],
+        name: `HTML page is served without the 'X-UA-Compatible' header, but it's required by the targeted browsers`,
+        reports: [{ message: `'x-ua-compatible' header was not specified` }],
+        serverConfig: { '/': '' }
+    }
+];
+
+const testsForMetaTagWithBlankTargetBrowsersConfig: Array<IRuleTest> = [
+    {
+        name: `'X-UA-Compatible' meta tag is not specified, but it's required by the targeted browsers`,
+        reports: [{ message: `No 'x-ua-compatible' meta tag was specified` }],
         serverConfig: { '/': { headers: { 'X-UA-Compatible': 'ie=edge' } } }
-    },
+    }
+];
+
+const testsForTargetBrowsersConfig: Array<IRuleTest> = [
     {
-        name: `'X-UA-Compatible' meta tag is not specified but the targeted browsers don't support document modes`,
-        reports: [{ message: `Meta tag is not needed` }],
-        serverConfig: { '/': { content: generateHTMLPageWithMetaTag() } }
+        name: `HTML page is served with both 'X-UA-Compatible' header and meta tag, but they are not required by the targeted browsers`,
+        reports: [
+            { message: `'x-ua-compatible' header is not needed` },
+            { message: `Meta tag is not needed` }
+        ],
+        serverConfig: {
+            '/': {
+                content: generateHTMLPageWithMetaTag(),
+                headers: { 'X-UA-Compatible': 'ie=edge' }
+            }
+        }
     }
 ];
 
 ruleRunner.testRule(ruleName, testsForDefaults);
 ruleRunner.testRule(ruleName, testsForRequireMetaTagConfig, { ruleOptions: { requireMetaTag: true } });
+ruleRunner.testRule(ruleName, testsForHeaderWithBlankTargetBrowsersConfig, { browserslist: [] });
+ruleRunner.testRule(ruleName, testsForMetaTagWithBlankTargetBrowsersConfig, {
+    browserslist: [],
```

Comment:
@alrra I think we should decide first what to do with what @qzhou1607 has pointed out in that comment. I think we should use browserlist's default behavior and don't make any assumption in our rules. The reason is that different rules could implement different "default behaviors", tracking them and making sure all are consistent is more complicated than just having a single place and update it there in the future.